frontend/javascript/basket.js

```diff
@@ -0,0 +1,86 @@
+function removeCartItem() {
+    let cartRemove_btns = document.querySelectorAll('.cart-remove');
+
+    cartRemove_btns.forEach((btn) => {
+        btn.addEventListener('click', function () {
+            btn.parentElement.remove();
+            updateTotal();
+        });
+    });
+}
+
+removeCartItem();
+
+function updateTotal() {
+    let cartBoxes = document.querySelectorAll('.cart-box');
+    let totalPrice = document.querySelector('.total__price');
+    let total = 0;
+
+    cartBoxes.forEach((cartBox) => {
+        let priceElement = cartBox.querySelector('.cart-box__product_price');
+        let price = parseFloat(priceElement.innerHTML.replace('£', ''));
+        let quantity = cartBox.querySelector('.cart-box__product-quantity').innerHTML;
+        total += price * quantity;
+    });
+
+    total = total.toFixed(2);
+
+    totalPrice.innerHTML = '£' + total;
+}
+
+function changeQuantity() {
+    let cartBoxes = document.querySelectorAll('.cart-box');
+    let decreaseQtyImg = document.querySelector('.cart-box-btn__disabled ');
+    cartBoxes.forEach((cartBox) => {
+        let currentQuantityElement = document.querySelector('.cart-box__product-quantity');
+        let currentQuantity = Number(
```

Comment:
I know we've already talked about that and that it may be on your todo list, but adding a note just for documentation. Rather than keeping the state of the quantity in the html, could we keep it as an object in memory alternatively in the session storage?

Reply:
Quantity now stored in sessionStorage